internal/client/util.go

```diff
@@ -3,8 +3,45 @@ package client
 import (
 	"fmt"
 	"net/http"
+	"strings"
+
+	"github.com/google/uuid"
 )
 
+// getAccountScopedURL constructs a URL for an account-scoped route
+func getAccountScopedURL(endpoint string, accountID uuid.UUID, route string) string {
+	var sb strings.Builder
+	sb.WriteString(endpoint)
+	sb.WriteString("/accounts/")
+	sb.WriteString(accountID.String())
+	sb.WriteRune('/')
+	sb.WriteString(route)
```

Comment:
probably revealing my ignorance here, but why not just construct through `sprintf`?

Reply:
No strong reason, really, just preference. I think this should perform marginally better than `fmt.Sprintf` because it avoids some necessary formatting of the format string and would support preallocating the buffer.

This approach is marginally beneficial for conditionals, since the alternative is to maintain two similar Sprintf statements. Arguably that would be more readable, I'm not opposed to refactoring this later 🤷